packages/meditrak-server/src/tests/routes/surveyResponse/testOutdatedStatusUpdate.js

```diff
@@ -0,0 +1,495 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+/* eslint-disable camelcase */
+
+import { expect } from 'chai';
+
+import { buildAndInsertSurveys, generateTestId } from '@tupaia/database';
+import { SurveyResponseChangeHandler } from '../../../database/SurveyResponseChangeHandler/SurveyResponseChangeHandler';
+
+const REFRESH_DEBOUNCE_TIME = 50; // short debounce time so tests run more quickly
+
+const buildSurvey = (id, periodGranularity) => {
+  const code = `Test_${periodGranularity || 'no_granularity'}`;
+
+  return {
+    id,
+    code,
+    period_granularity: periodGranularity,
+    questions: [{ code: `${code}1`, type: 'Number' }],
+  };
+};
+
+const yearlySurveyId = generateTestId();
+const quarterlySurveyId = generateTestId();
+const monthlySurveyId = generateTestId();
+const weeklySurveyId = generateTestId();
+const dailySurveyId = generateTestId();
+const nonPeriodicSurveyId = generateTestId();
+
+const SURVEYS = {
+  [yearlySurveyId]: buildSurvey(yearlySurveyId, 'yearly'),
+  [quarterlySurveyId]: buildSurvey(quarterlySurveyId, 'quarterly'),
+  [monthlySurveyId]: buildSurvey(monthlySurveyId, 'monthly'),
+  [weeklySurveyId]: buildSurvey(weeklySurveyId, 'weekly'),
+  [dailySurveyId]: buildSurvey(dailySurveyId, 'daily'),
+  [nonPeriodicSurveyId]: buildSurvey(nonPeriodicSurveyId),
+};
+
+export const testOutdatedStatusUpdate = app => {
```

Comment:
I chose to add the majority of test cases for this feature in the CUD routes for surveyResponse, instead of doing so in the model level. The latter would cover a bigger spectrum of use cases (eg changes in migrations), but I still felt that testing in the api level is more appropriate.

We could have additional tests in the model level, but I thought that would be over-testing. Plus, one could consider the fact that change handling logic is also available in the model level an implementation detail.